Aplib.Tests/Desire/GoalStructureTests.cs

```diff
@@ -34,14 +31,10 @@ public void FirstOfGoalStructure_WhenAllGoalsFail_ShouldReturnFailure()
     public void FirstOfGoalStructure_WhenDisposing_ShouldDisposeChildren()
     {
         // Arrange
-        Mock<IGoalStructure<BeliefSet>> goalStructure1 = new();
-        Mock<IGoalStructure<BeliefSet>> goalStructure2 = new();
+        Mock<IGoalStructure<IBeliefSet>> goalStructure1 = new();
+        Mock<IGoalStructure<IBeliefSet>> goalStructure2 = new();
 
-        Mock<FirstOfGoalStructure<BeliefSet>> firstOfGoalStructure =
-            new(new List<IGoalStructure<BeliefSet>> { goalStructure1.Object, goalStructure2.Object })
-            {
-                CallBase = true
-            };
+        Mock<FirstOfGoalStructure<IBeliefSet>> firstOfGoalStructure = new([goalStructure1.Object, goalStructure2.Object]);
```

Comment:
CallBase has to be set to true in the constructor, otherwise we won't get any coverage for the dispose method, and then this test could be removed.